src/Hl7.Fhir.Specification/Specification/Snapshot/ElementDefnMerger.cs

```diff
@@ -244,6 +245,58 @@ void onConstraint(Element snap)
                     return snap;
             }
 
+            // The diamond problem is especially painful for min/max -
+            // most datatypes roots have a cardinality of 0..*, so
+            // a non-repeating element referring to a profiled datatype
+            // (with 0..*) onto an element which needs to be expanded
+            // and which has no diff constraints on max will get the 
+            // max from the base -> a non-repeating element will become repeating.
+            internal FhirString mergeMax(FhirString snap, FhirString diff)
+            {
+                // [WMR 20160718] Handle snap == null
+                // if (!diff.IsNullOrEmpty() && !diff.IsExactly(snap))
+                // if (!diff.IsNullOrEmpty() && (snap == null || !diff.IsExactly(snap)))
+                if (!diff.IsNullOrEmpty() && !snap.IsNullOrEmpty() && !diff.IsExactly(snap))
+                {
+                    if (diff.Value == "*")
+                        return snap;
+
+                    // Now, diff has a numeric limit
+                    // So, if snap has no limit, take the diff
+                    if(snap.Value == "*")
+                    {
+                        var result = (FhirString)diff.DeepCopy();
```

Comment:
This part is coming back 3 times, perhaps create an inline function for this:

```
FhirString deepCopy(FhirString elt)
{
   var result = (FhirString)elt.DeepCopy();
   onConstraint(result);
   return result;
}
```

Reply:
@marcovisserFurore , I tried that, but unfortunately, ElementDefnMerger is a struct - and local functions cannot create a closure over instance variables like `onConstraint` within a struct. I guess Michel made this a struct since it's being created quite often.

Anyway, since we both don't like copy & paste - I turned it into a full blown member function ;-)